File: src/com/novell/android/yastroid/ServerEditActivity.java
package com.novell.android.yastroid;

import static com.novell.android.yastroid.YastroidOpenHelper.*;

import java.net.URI;
import java.util.ArrayList;

import android.app.Activity;
import android.content.ContentValues;
import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class ServerEditActivity extends Activity {

	private SQLiteDatabase database;
	private YastroidOpenHelper dbhelper;
	private Button saveButton;
	private Server s;

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.serveredit);

		dbhelper = new YastroidOpenHelper(this);

		Intent serverIntent = getIntent();
		Bundle b = serverIntent.getExtras();
		getServer(b.getInt("SERVER_ID"));
		fillLayout( );

		this.saveButton = (Button)this.findViewById(R.id.button_save_server);
		this.saveButton.setOnClickListener(new OnClickListener() {
			@Override
			public void onClick(View v) {
				if(saveServer()) {
					finish();
				}
			}
		});
	}

	private void getServer(long serverId) {
		database = dbhelper.getReadableDatabase();
		Cursor sc;
		try {
				sc = database.query(SERVERS_TABLE_NAME, new String[] {
						"_id", "name", "scheme", "hostname", "port", "user", "pass", "grp" },
						SERVERS_ID + "=" + serverId, null, null, null, null);

			sc.moveToFirst();
			s = new Server(sc.getInt(0), sc.getString(1), sc.getString(2), sc
							.getString(3), sc.getInt(4), sc.getString(5), sc
							.getString(6), sc.getInt(7));
			sc.close();
			database.close();
		} catch (Exception e) {
			Log.e("BACKGROUND_PROC", e.getMessage());
		}
	}
	
	private void fillLayout() {
		((EditText)findViewById(R.id.edit_server_name)).setText(s.getName());
		((EditText)findViewById(R.id.edit_server_host)).setText(s.getScheme() + "://"+ s.getHostname());
		((EditText)findViewById(R.id.edit_server_port)).setText(Integer.toString(s.getPort()));
		((EditText)findViewById(R.id.edit_server_user)).setText(s.getUser());
	}

	private boolean saveServer() {
		database = dbhelper.getWritableDatabase();
		boolean result = false;
		try {
			
			URI uri = new URI(((EditText)findViewById(R.id.edit_server_host)).getText().toString());
			String name = ((EditText)findViewById(R.id.edit_server_name)).getText().toString();
			String scheme = null;
			String host = uri.getHost();
			String port = ((EditText)findViewById(R.id.edit_server_port)).getText().toString();
			String user = ((EditText)findViewById(R.id.edit_server_user)).getText().toString();
			String pass = ((EditText)findViewById(R.id.edit_server_pass)).getText().toString();
			
			if (uri.getScheme() == null) {
				scheme = "http";
				host = uri.getSchemeSpecificPart();
			} else {
				scheme = uri.getScheme();
			}
			
			if (name.length() == 0 || host.length() == 0 || port.length() == 0 || user.length() == 0) {
				Toast.makeText(ServerEditActivity.this, "One or more fields are empty", Toast.LENGTH_SHORT).show();
				return false;
			}

			ContentValues values = new ContentValues();
			values.put(SERVERS_NAME, name);
			values.put(SERVERS_SCHEME, scheme);
			values.put(SERVERS_HOST, host);
			values.put(SERVERS_PORT, port);
			values.put(SERVERS_USER, user);
			if(pass.length() == 0 )
				values.put(SERVERS_PASS, s.getPass());
			else
				values.put(SERVERS_PASS, pass);
			values.put(SERVERS_GROUP, s.getGroupId());
			
			database.update(SERVERS_TABLE_NAME, values, SERVERS_ID + "=" + s.getId(), null);
			database.close();
			Log.i("ARRAY", "WebYaST server " + name + " has been updated.");
			result = true;
		} catch (Exception e) {
			Log.e("BACKGROUND_PROC", e.getMessage());
		}
		Toast.makeText(ServerEditActivity.this, "Server Updated", Toast.LENGTH_SHORT).show();
		return result;
	}
}

File: mono/YaSTroid/YaSTroid/ServerEditActivity.cs
using System;
using Android.App;
using Android.Content;
using Android.Database;
using Android.Database.Sqlite;
using Android.OS;
using Android.Util;
using Android.Widget;
using Java.Net;

namespace YaSTroid
{
	[Activity (Label = "ServerActivity")]
	public class ServerEditActivity : Activity
	{

		private SQLiteDatabase database;
		private YastroidOpenHelper dbhelper;
		private Button saveButton;
		private Server s;

		protected override void OnCreate(Bundle savedInstanceState)
		{
			base.OnCreate(savedInstanceState);
			SetContentView(Resource.Layout.serveredit);

			dbhelper = new YastroidOpenHelper(this);

			Intent serverIntent = Intent;
			Bundle b = serverIntent.Extras;
			getServer(b.GetInt("SERVER_ID"));
			fillLayout( );

			this.saveButton = FindViewById<Button>(Resource.Id.button_save_server);
			this.saveButton.Click += (sender, e) => {
				if(saveServer())
					Finish();
			};
		}

		private void getServer(long serverId) {
			database = dbhelper.ReadableDatabase;
			ICursor sc;
			try {
				sc = database.Query(YastroidOpenHelper.SERVERS_TABLE_NAME, new string[] {
							"_id", "name", "scheme", "hostname", "port", "user", "pass", "grp" },
					YastroidOpenHelper.SERVERS_ID + "=" + serverId, null, null, null, null);

				sc.MoveToFirst();
				s = new Server(sc.GetInt(0), sc.GetString(1), sc.GetString(2), sc
					.GetString(3), sc.GetInt(4), sc.GetString(5), sc
								.GetString(6), sc.GetInt(7));
				sc.Close();
				database.Close();
			} catch (Exception e) {
				Log.Error("BACKGROUND_PROC", e.Message);
			}
		}
		
		private void fillLayout()
		{
			FindViewById<EditText>(Resource.Id.edit_server_name).Text = (s.getName());
			FindViewById<EditText>(Resource.Id.edit_server_host).Text = (s.getScheme() + "://"+ s.getHostname());
			FindViewById<EditText>(Resource.Id.edit_server_port).Text = (s.getPort().ToString ());
			FindViewById<EditText>(Resource.Id.edit_server_user).Text = (s.getUser());
		}

		private bool saveServer()
		{
			database = dbhelper.WritableDatabase;
			bool result = false;
			try {
				
				URI uri = new URI(FindViewById<EditText>(Resource.Id.edit_server_host).Text);
				string name = FindViewById<EditText>(Resource.Id.edit_server_name).Text;
				string scheme = null;
				string host = uri.Host;
				string port = FindViewById<EditText>(Resource.Id.edit_server_port).Text;
				string user = FindViewById<EditText>(Resource.Id.edit_server_user).Text;
				string pass = FindViewById<EditText>(Resource.Id.edit_server_pass).Text;
				
				if (uri.Scheme == null) {
					scheme = "http";
					host = uri.SchemeSpecificPart;
				} else {
					scheme = uri.Scheme;
				}
				
						if (name.Length == 0 || host.Length == 0 || port.Length == 0 || user.Length == 0) {
							Toast.MakeText(this, "One or more fields are empty", ToastLength.Short).Show();
					return false;
				}

				ContentValues values = new ContentValues();
				values.Put(YastroidOpenHelper.SERVERS_NAME, name);
				values.Put(YastroidOpenHelper.SERVERS_SCHEME, scheme);
				values.Put(YastroidOpenHelper.SERVERS_HOST, host);
				values.Put(YastroidOpenHelper.SERVERS_PORT, port);
				values.Put(YastroidOpenHelper.SERVERS_USER, user);
				if(pass.Length == 0 )
					values.Put(YastroidOpenHelper.SERVERS_PASS, s.getPass());
				else
					values.Put(YastroidOpenHelper.SERVERS_PASS, pass);
					values.Put(YastroidOpenHelper.SERVERS_GROUP, s.getGroupId());
				
				database.Update(YastroidOpenHelper.SERVERS_TABLE_NAME, values, YastroidOpenHelper.SERVERS_ID + "=" + s.getId(), null);
						database.Close();
						Log.Info("ARRAY", "WebYaST server " + name + " has been updated.");
				result = true;
			} catch (Exception e) {
							Log.Error("BACKGROUND_PROC", e.Message);
			}
						Toast.MakeText(this, "Server Updated", ToastLength.Short).Show();
			return result;
		}
	}
}
